src/array/reverse.js

```diff
@@ -0,0 +1,14 @@
+define(function () {
+
+    /**
+     * Returns a copy of the array in reversed order.
+     */
+    function reverse(array) {
+        var copy = array.slice();
```

Comment:
I could use `mout/lang/clone` but opted for the shorter version to skip unnecessary dependencies. Do we want to keep this DRY and add `clone()`?